File: ios/RNPDFEditorViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>
#import <React/RCTLog.h>

@interface RCT_EXTERN_MODULE(RNPDFEditorViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(options, NSDictionary *)

RCT_EXPORT_VIEW_PROPERTY(onSavePDF, RCTDirectEventBlock)

RCT_EXTERN_METHOD(undoAction:(nonnull NSNumber *)node)

RCT_EXTERN_METHOD(clearAction:(nonnull NSNumber *)node)

RCT_EXTERN_METHOD(saveAction:(nonnull NSNumber *)node)

@end

File: ios/RNPDFEditorViewManager.swift
import React

@objc(RNPDFEditorViewManager)
class RNPDFEditorViewManager: RCTViewManager {

    override func view() -> UIView! {
        return ContainerView()
    }

    @objc override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    @objc func setEditMode(_ node:NSNumber, isEdit: NSNumber) {
      DispatchQueue.main.async {
          guard let component = self.bridge.uiManager.view(
              forReactTag: node
          ) as? ContainerView else {
              print("RNPDFEditor: Cannot find Native UIView with tag", node)
              return
          }
          component.setEditMode(isEdit.boolValue)
      }
    }

    @objc func undoAction(_ node:NSNumber) {
      DispatchQueue.main.async {
          guard let component = self.bridge.uiManager.view(
              forReactTag: node
          ) as? ContainerView else {
              print("RNPDFEditor: Cannot find Native UIView with tag", node)
              return
          }
          component.undo()
      }
    }

    @objc func clearAction(_ node:NSNumber) {
      DispatchQueue.main.async {
          guard let component = self.bridge.uiManager.view(
              forReactTag: node
          ) as? ContainerView else {
              print("RNPDFEditor: Cannot find Native UIView with tag", node)
              return
          }
          component.clear()
      }
    }

    @objc func saveAction(_ node:NSNumber) {
      DispatchQueue.main.async {
          guard let component = self.bridge.uiManager.view(
              forReactTag: node
          ) as? ContainerView else {
              print("RNPDFEditor: Cannot find Native UIView with tag", node)
              return
          }
          component.save()
      }
    }
}
